test/spec/adapters/jcm_spec.js

```diff
@@ -0,0 +1,246 @@
+            expect(parsedBidUrlQueryString).to.have.property('bids');
+
+            var bidObjArr = JSON.parse(parsedBidUrlQueryString.bids);
+            expect(bidObjArr).to.have.property('bids');
+            var bidObj1 = bidObjArr.bids[0];
+
+            expect(bidObj1).to.have.property('adSizes').and.to.equal('300x250');
+            expect(bidObj1).to.have.property('siteId').and.to.equal('3608');
+            expect(bidObj1).to.have.property('callbackId').and.to.equal('3c9408cdbf2f68');
+
+            var bidObj2 = bidObjArr.bids[1];
+
+            expect(bidObj2).to.have.property('adSizes').and.to.equal('728x90');
+            expect(bidObj2).to.have.property('siteId').and.to.equal('3608');
+            expect(bidObj2).to.have.property('callbackId').and.to.equal('3c9408cdbf2f69');
+
+      });
+
+  });
+
+  describe('handling of the callback response', function () {
+
+      pbjs._bidsReceived = [];
+      pbjs._bidsRequested = [];
+      pbjs._adsReceived = [];
+
+      var params = {
+                bidderCode: 'jcm',
+                bidder: 'jcm',
+                bidderRequestId: '2068db3c904101',
+                bids: [
+                        {
+                          bidId: '3c9408cdbf2f68',
+                          sizes: [[300, 250]],
+                          bidder: 'jcm',
+                          params: { siteId: '3608', adSizes:'300x250' },
+                          requestId: '10b327aa396609',
+                          placementCode: '/19968336/header-bid-tag-0'
+                         },
+                         {
+                          bidId: '3c9408cdbf2f69',
+                          sizes: [[728, 90]],
+                          bidder: 'jcm',
+                          params: { siteId: '3608', adSizes:'728x90' },
+                          requestId: '10b327aa396610',
+                          placementCode: '/19968336/header-bid-tag-1'
+                         }
+
+                ]
+            };
+
+
+       var response='{"bids":[{"width":300,"cpm":3,"ad":"%3Cimg+src%3D%22http%3A%2F%2Fmedia.adfrontiers.com%2Fimgs%2Fpartnership_300x250.png%22%3E","callbackId":"3c9408cdbf2f68","height":250},{"width":728,"cpm":0,"ad":"%3Cimg+src%3D%22http%3A%2F%2Fmedia.adfrontiers.com%2Fimgs%2Fpartnership_728x90.png%22%3E","callbackId":"3c9408cdbf2f69","height":90}]}';
+
+
+
+       it('callback function should exist', function () {
+            expect(pbjs.processJCMResponse).to.exist.and.to.be.a('function');
+       });
+
+        it('bidmanager.addBidResponse should be called twice with correct arguments', function () {
+
+       var stubAddBidResponse = sinon.stub(bidmanager, 'addBidResponse');
+      
+       adapter().callBids(params);
+
+       var adUnits = new Array();
+       var unit = new Object();
+       unit.bids = [params];
+       unit.code = '/19968336/header-bid-tag';
+       unit.sizes=[[300,250],[728,90]];
+       adUnits.push(unit);
+
+       pbjs._bidsRequested = [params];
```

Comment:
This line causes some of the other tests that use `pbjs._bidsRequested` to fail. If you change this line to add `param` to the existing data with `pbjs._bidsRequested.push(params);` this will let other tests pass too.